test/functional/pdo_sqlsrv/pdo_ae_insert_pdoparam_char_size.phpt

```diff
@@ -0,0 +1,187 @@
+--TEST--
+Test for inserting encrypted data into char types columns with different sizes
+--DESCRIPTION--
+Test conversions between different char types of different sizes
+With or without Always Encrypted, implicit conversion works if:
+1. From input of PDO::PARAM_BOOL to a any char column
+2. From input of PDO::PARAM_INT to a any char column
+3. From input of PDO::PARAM_STR to a any char column
+4. From input of PDO::PARAM_LOB to a any char column
+--SKIPIF--
+<?php require('skipif_mid-refactor.inc'); ?>
+--FILE--
+<?php
+require_once("MsCommon_mid-refactor.inc");
+require_once("AEData.inc");
+
+$dataTypes = array("char", "varchar", "varchar(max)");
+$lengths = array(1, 8, 64, 512, 4096, 8000);
+
+try {
+    $conn = connect();
+    foreach ($dataTypes as $dataType) {
+        $maxcol = strpos($dataType, "(max)");
+        foreach ($lengths as $m) {
+            if ($maxcol !== false) {
+                $typeFull = $dataType;
+            } else {
+                $typeFull = "$dataType($m)";
+            }
+            echo "\nTesting $typeFull:\n";
+                
+            //create table containing char(m) or varchar(m) columns
+            $tbname = "test_" . str_replace(array('(', ')'), '', $dataType) . $m;
+            $colMetaArr = array(new ColumnMeta($typeFull, "c1", null, "randomized"));
+            createTable($conn, $tbname, $colMetaArr);
```

Comment:
because with AE off, a row has a maximum column size of 8000 bytes. For lengths of 4096 and 8000 (see line 18), a table with two columns of this size cannot be created.